src/st/no1/unk_381E8.c

```diff
@@ -333,8 +334,29 @@ void func_us_801B8B00(Entity* self) {
 
 INCLUDE_ASM("st/no1/nonmatchings/unk_381E8", func_us_801B8D30);
 
-// entering doppleganger's room
```

Comment:
I think this comment was a red herring from debugging because the doppleganger room moves into RBO4 and this function probably sits at a similar address in memory. Given the decompiled code references the elevator flag, and the code directly above this is all elevator related I think it's safe to say this is probably part of an elevator code file.